sound/soc/sof/intel/mtl.c

```diff
@@ -431,6 +433,12 @@ static int mtl_dsp_cl_init(struct snd_sof_dev *sdev, int stream_tag, bool imr_bo
 
 	dev_dbg(sdev->dev, "Primary core power up successful\n");
 
+	/* set enabled cores mask and increment ref count for cores in init_core_mask */
+	sdev->enabled_cores_mask |= chip->init_core_mask;
+	mask = sdev->enabled_cores_mask;
+	for_each_set_bit(i, &mask, SOF_MAX_DSP_NUM_CORES)

```

Comment:
@RanderWang we can simplify this as we know for a fact that only core 0 is the only core that will be enabled after ROM init

Reply:
we do that in other platforms, no?

Reply:
yes, only because in cavs we had the difference for ex cnl has all cores enabled after init and TGL had only core 0. But for MTL, why unnecessarily complicate it? We know that core 0 is the only possibility

Reply:
let's simplify it.